AK/BigIntBase.h

```diff
@@ -0,0 +1,595 @@
+{
+    if (!__builtin_is_constant_evaluated()) {
+#if ARCH(X86_64)
+        unsigned long long output;
+        carry = __builtin_ia32_addcarryx_u64(carry, word1, word2, &output);
+        return static_cast<NativeWord>(output);
+#else
+#    if defined(AK_COMPILER_CLANG)
+        NativeWord ncarry, output;
+        if constexpr (SameAs<NativeWord, unsigned int>)
+            output = __builtin_addc(word1, word2, carry, reinterpret_cast<unsigned int*>(&ncarry));
+        else if constexpr (SameAs<NativeWord, unsigned long>)
+            output = __builtin_addcl(word1, word2, carry, reinterpret_cast<unsigned long*>(&ncarry));
+        else if constexpr (SameAs<NativeWord, unsigned long long>)
+            output = __builtin_addcll(word1, word2, carry, reinterpret_cast<unsigned long long*>(&ncarry));
+        else
+            VERIFY_NOT_REACHED();
+        carry = ncarry;
+        return output;
+#    endif
+#endif
+    }
+    // Note: This is usually too confusing for both GCC and Clang.
+    NativeWord output;
+    bool ncarry = __builtin_add_overflow(word1, word2, &output);
+    if (carry) {
+        ++output;
+        if (output == 0)
+            ncarry = true;
+    }
+    carry = ncarry;
+    return output;
+}
+
+ALWAYS_INLINE constexpr NativeWord sub_words(NativeWord word1, NativeWord word2, bool& carry)
+{
+    if (!__builtin_is_constant_evaluated()) {
+#if ARCH(X86_64)
+        unsigned long long output;
+#    if defined(AK_COMPILER_CLANG)
+        // This is named __builtin_ia32_sbb_u64 in G++. *facepalm*
+        carry = __builtin_ia32_subborrow_u64(carry, word1, word2, &output);
+#    else
+        carry = __builtin_ia32_sbb_u64(carry, word1, word2, &output);
+#    endif
+        return static_cast<NativeWord>(output);
+#else
+#    if defined(AK_COMPILER_CLANG)
+        NativeWord ncarry, output;
+        if constexpr (SameAs<NativeWord, unsigned int>)
+            output = __builtin_subc(word1, word2, carry, reinterpret_cast<unsigned int*>(&ncarry));
+        else if constexpr (SameAs<NativeWord, unsigned long>)
+            output = __builtin_subcl(word1, word2, carry, reinterpret_cast<unsigned long*>(&ncarry));
+        else if constexpr (SameAs<NativeWord, unsigned long long>)
+            output = __builtin_subcll(word1, word2, carry, reinterpret_cast<unsigned long long*>(&ncarry));
+        else
+            VERIFY_NOT_REACHED();
+        carry = ncarry;
+        return output;
+#    endif
+#endif
+    }
+    NativeWord output;
+    bool ncarry = __builtin_sub_overflow(word1, word2, &output);
+    if (carry) {
+        if (output == 0)
+            ncarry = true;
+        --output;
+    }
+    carry = ncarry;
+    return output;
+}
+
+// Calculate ((divident1 << word_size) + divident0) / divisor. Quotient should be guaranteed to fit into NativeWord.
+ALWAYS_INLINE constexpr NativeWord div_mod_words(NativeWord divident0, NativeWord divident1, NativeWord divisor, NativeWord& remainder)
```

Comment:
`divident` is an obsolete synonym for `dividend`. Applies to more than just the first commit.

https://books.google.com/ngrams/graph?content=divident%2C+dividend&year_start=1800&year_end=2019&corpus=en-2019&smoothing=3